File: CP(Competetive Programming)/1.Practice/0. vscode/CodingNinja/Insertion Sort/InsertionSort.py
from os import *
from sys import *
from collections import *
from math import *

from typing import List


def insertionSort(n: int, arr: List[int]) -> None:
    for i in range(n):
        if (i != n-1):
            for j in range(i+1, 0, -1):
                if (arr[i] > arr[j]):
                    arr[i], arr[j] = arr[j], arr[i]
    print(arr)


arr = [1, 4, 2]
insertionSort(3, arr)
print(arr)

File: CP(Competetive Programming)/1.Practice/0. vscode/CodingNinja/Insertion Sort/InsertionSort.java
public class InsertionSort {
    public static void sort(int n, int[] arr) {
        for (int i = 0; i < n - 1; i++) {
            int j = i;
            int key = arr[i + 1];
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j = j - 1;
            }
            arr[j + 1] = key;
        }
    }

    public static void main(String[] args) {
        int n = 5;
        int arr[] = { 23, 1, 10, 5, 2 };
        sort(n, arr);
        for (int val : arr) {
            System.out.println(val);
        }
        // System.out.println(arr);
    }
}